code/modules/mining/ores_coins.dm

```diff
@@ -304,18 +304,31 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 /obj/item/coin
 	icon = 'icons/obj/economy.dmi'
 	name = "coin"
-	icon_state = "coin__heads"
+	icon_state = "coin"
 	flags_1 = CONDUCT_1
 	force = 1
 	throwforce = 2
 	w_class = WEIGHT_CLASS_TINY
+	custom_materials = list(/datum/material/iron = 400)
 	var/string_attached
 	var/list/sideslist = list("heads","tails")
-	var/cmineral = null
 	var/cooldown = 0
-	var/value = 1
+	var/value
 	var/coinflip
 
+/obj/item/coin/Initialize()
+	. = ..()
+	coinflip = pick(sideslist)
+	icon_state = "coin_[coinflip]"
+	pixel_x = rand(0,16)-8
+	pixel_y = rand(0,8)-8
+
+/obj/item/coin/set_custom_materials()
+	. = ..()

```

Comment:
add value = 0 to prevent the value from going up indefinitely when you change materials